enterprise/server/remote_execution/persistentworker/persistentworker.go

```diff
@@ -79,8 +80,19 @@ func Start(ctx context.Context, workspace *workspace.Workspace, container contai
 		stdinWriter:  stdinWriter,
 		stdoutReader: bufio.NewReader(stdoutReader),
 	}
+
+	// Use the same size limit as regular actions to prevent OOM from persistent worker stderr
+	if *commandutil.StdOutErrMaxSize > 0 {
+		w.stderr = *lockingbuffer.NewWithMaxSize(*commandutil.StdOutErrMaxSize)
+	} else {
+		w.stderr = *lockingbuffer.New()
+	}
```

Comment:
The pointer dereferencing in `w.stderr = *lockingbuffer.New()` is unusual - I think generally, if a constructor function returns a pointer, that means the returned object is intended to be used as a reference, so we should not de-reference it. It'd be better to change the `stderr` field to be a pointer type, IMO.